IPython/core/displayhook.py

```diff
@@ -83,12 +84,18 @@ def check_for_underscore(self):
     def quiet(self):
         """Should we silence the display hook because of ';'?"""
         # do not print output if input ends in ';'
-        try:
-            cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
-            return cell.rstrip().endswith(';')
-        except IndexError:
-            # some uses of ipshellembed may fail here
-            return False
+
+	cell = self.shell.history_manager.input_hist_parsed[self.prompt_count]
```

Comment:
The code that was here before catches `IndexError`, in case `input_hist_parsed` doesn't have that many entries. I'm not sure if that can actually still happen, but we should probably still catch that error, just to be on the safe side.